### PR TITLE
Add Feature: Dynamic redirects

### PR DESCRIPTION
**Pull request by litewarp**
Implementation of dynamic redirects in discussion at #57.

### CODE REVIEW

File: src/__tests__/withAuthUser.test.js [hunk 1]
```diff
@@ -334,6 +334,77 @@ describe('withAuthUser: rendering/redirecting', () => {
...
+      whenUnauthedAfterInit: AuthAction.REDIRECT_TO_LOGIN,
+      whenAuthed: AuthAction.RENDER,
+    })(MockComponent)
+    render(
+      <MockCompWithUser
+        serializedAuthUser={MockSerializedAuthUser}
+        message="How are you?"
+      />
+    )
+    expect(propsSpy.ctx).not.toBeTruthy()
```
Comment (kmjennison) [inline src/__tests__/withAuthUser.test.js:403]: To be more precise, use `toBeUndefined()`.

File: src/__tests__/withAuthUser.test.js [hunk 2]
```diff
@@ -429,6 +500,79 @@ describe('withAuthUser: rendering/redirecting', () => {
...
+      whenUnauthedAfterInit: AuthAction.RENDER,
+      whenAuthed: AuthAction.REDIRECT_TO_APP,
+    })(MockComponent)
+    render(
+      <MockCompWithUser
+        serializedAuthUser={MockSerializedAuthUser}
+        message="How are you?"
+      />
+    )
+    expect(ctxSpy).not.toBeTruthy()
```
Comment (kmjennison) [inline src/__tests__/withAuthUser.test.js:570]: Use `toBeUndefined()`.

File: src/__tests__/withAuthUserTokenSSR.test.js [hunk 3]
```diff
@@ -418,6 +418,25 @@ describe('withAuthUserTokenSSR: redirect and composed prop logic', () => {
     })
   })
 
+  it('redirects to the provided function login URL when the user is not auth and auth *is* required', async () => {
```
Comment (kmjennison) [inline src/__tests__/withAuthUserTokenSSR.test.js:421]: Typo: "when the user is not **authed**"

**Review by kmjennison — CHANGES_REQUESTED** [review]
This looks excellent. Thank you for yet another contribution and for the great tests!

Only recommending a few tiny changes, then happy to merge.

**Comment by litewarp** [comment]
> This looks excellent. Thank you for yet another contribution and for the great tests!

Thanks! Thank you for the great library.

**Review by kmjennison — APPROVED** [review]
LGTM! 🎊

**Pull request merged**